modules/animation/include/modules/animation/algorithm/animationrange.h

```diff
@@ -0,0 +1,62 @@
+/*********************************************************************************
+ *
+ * Inviwo - Interactive Visualization Workshop
+ *
+ * Copyright (c) 2020 Inviwo Foundation
+ * All rights reserved.
+ *
+ * Redistribution and use in source and binary forms, with or without
+ * modification, are permitted provided that the following conditions are met:
+ *
+ * 1. Redistributions of source code must retain the above copyright notice, this
+ * list of conditions and the following disclaimer.
+ * 2. Redistributions in binary form must reproduce the above copyright notice,
+ * this list of conditions and the following disclaimer in the documentation
+ * and/or other materials provided with the distribution.
+ *
+ * THIS SOFTWARE IS PROVIDED BY THE COPYRIGHT HOLDERS AND CONTRIBUTORS "AS IS" AND
+ * ANY EXPRESS OR IMPLIED WARRANTIES, INCLUDING, BUT NOT LIMITED TO, THE IMPLIED
+ * WARRANTIES OF MERCHANTABILITY AND FITNESS FOR A PARTICULAR PURPOSE ARE
+ * DISCLAIMED. IN NO EVENT SHALL THE COPYRIGHT OWNER OR CONTRIBUTORS BE LIABLE FOR
+ * ANY DIRECT, INDIRECT, INCIDENTAL, SPECIAL, EXEMPLARY, OR CONSEQUENTIAL DAMAGES
+ * (INCLUDING, BUT NOT LIMITED TO, PROCUREMENT OF SUBSTITUTE GOODS OR SERVICES;
+ * LOSS OF USE, DATA, OR PROFITS; OR BUSINESS INTERRUPTION) HOWEVER CAUSED AND
+ * ON ANY THEORY OF LIABILITY, WHETHER IN CONTRACT, STRICT LIABILITY, OR TORT
+ * (INCLUDING NEGLIGENCE OR OTHERWISE) ARISING IN ANY WAY OUT OF THE USE OF THIS
+ * SOFTWARE, EVEN IF ADVISED OF THE POSSIBILITY OF SUCH DAMAGE.
+ *
+ *********************************************************************************/
+#pragma once
+
+#include <modules/animation/animationmoduledefine.h>
+#include <modules/animation/datastructures/animationtime.h>
+
+#include <algorithm>
+#include <iterator>
+
+namespace inviwo {
+
+namespace animation {
+
+/*
+ * Returns iterators for elements in the range [from, to].
+ * Deals with cases where from > to.
+ * @return One iterator to first item >= min(from, to) and one iterator to item > max(from, to)
+ */
+template <typename Iterator>
+auto getRange(Iterator begin, Iterator end, Seconds from, Seconds to)
+    -> std::tuple<decltype(begin), decltype(end)> {
+    auto first = std::min(from, to);
+    auto last = std::max(from, to);
+    // 'fromIt' will be the first item with a time larger than or equal to 'first'
+    auto fromIt = std::lower_bound(begin, end, first,
```

Comment:
can't you just use equal range for this? https://en.cppreference.com/w/cpp/algorithm/equal_range

Reply:
Hmm, yes. I looked at it but I seem too have disregarded it too quickly

Reply:
On a closer look it seems as if it is easier to separate the two since we are not comparing with a value as equal_range intends to do.